packages/did/src/DidDetails/FullDidDetails.utils.ts

```diff
@@ -28,10 +29,24 @@ const methodMapping: Record<string, VerificationKeyRelationship | undefined> = {
   web3Names: 'authentication',
 }
 
-export function getKeyRelationshipForExtrinsic(
-  extrinsic: Extrinsic
+function getKeyRelationshipForCall<A extends AnyTuple = AnyTuple>(
+  call: CallBase<A>
 ): VerificationKeyRelationship | undefined {
-  const { section, method } = extrinsic.method
+  const { section, method } = call
+
+  // get the VerificationKeyRelationship of a batched call
+  if (
+    section === 'utility' &&
+    (method === 'batch' || method === 'batchAll' || method === 'forceBatch') &&
+    call.args[0].toRawType() === 'Vec<Call>'
+  ) {
+    // map all calls to their VerificationKeyRelationship and deduplicate the items
+    return (call.args[0] as unknown as Array<CallBase<A>>)
+      .map((innerCall: CallBase<A>) => {
+        return getKeyRelationshipForCall(innerCall)
+      })
```

Comment:
Is this more elegant?
```suggestion
      .map(getKeyRelationshipForCall)
```